test/template-tests/testcases/CreateUIDefinition/Allowed-Values-Should-Actually-Be-Allowed.test.ps1

```diff
@@ -0,0 +1,36 @@
+﻿param(
+[Parameter(Mandatory=$true)]
+[PSObject]
+$CreateUIDefinitionObject,
+
+[Parameter(Mandatory=$true)]
+[Collections.IDictionary]
+$MainTemplateParameters
+)
+
+$allowedValues = $CreateUIDefinitionObject | 
+    Find-AzureRMTemplate -Key allowedValues -Value * -Like
+
+foreach ($av in $allowedValues) {
+    $parent = $av.ParentObject[0]
+
+    $MainTemplateParam = $MainTemplateParameters[$parent.Name]
+
+    if (-not $MainTemplateParam) {

```

Comment:
If you're referring to the check that the parameter exists in main template - I'd prefer this to provide a useful message if the parameter is missing, rather than a weird one (because -TargetObject would then be null, which would be less than helpful)

Reply:
Got it - so let's add a comment here so that it's clear that this part of the code is only to help avoid a generic PS message when we try to parse allowedValues